HealthcareSystem/GraphicalEditor/Models/MapObjectRelated/MapObjectDoor.cs

```diff
@@ -0,0 +1,114 @@
+{
+    public class MapObjectDoor
+    {
+        private Rectangle _rectangle;
+        private MapObjectMetrics _mapObjectMetrics;
+        private MapObjectDoorOrientation _mapObjectDoorOrientation;
+        private double _XShift;
+        private double _YShift;
+
+        private readonly double DOOR_WIDTH = 45;
+        private readonly double DOOR_HEIGHT = 10;
+
+        public MapObjectDoor(MapObjectDoorOrientation mapObjectDoorOrientations, double XShift = 0, double YShift = 0)
+        {
+            MapObjectDoorOrientation = mapObjectDoorOrientations;
+            this.XShift = XShift;
+            this.YShift = YShift;
+        }
+
+        public Rectangle GetDoor()
+        {
+            Rectangle = new Rectangle();
+            Rectangle.Fill = Brushes.DarkGreen;
+            Rectangle.Width = GetDoorWidth();
+            Rectangle.Height = GetDoorHeight();
+
+            Rectangle.SetValue(Canvas.LeftProperty, CalculateDoorX());
+            Rectangle.SetValue(Canvas.TopProperty, CalculateDoorY());
+
+            return Rectangle;
+        }
+
+        // Rotacija vrata - zameni sirinu i visinu ukoliko vrata trebaju da se postave vertikalno ( horizontalno su po default)
+        private double GetDoorWidth()
+        {
+            if (MapObjectDoorOrientation == MapObjectDoorOrientation.TOP || MapObjectDoorOrientation == MapObjectDoorOrientation.BOTTOM)
+                return DOOR_WIDTH;
+            else return DOOR_HEIGHT;
+        }
+
+        // Rotacija vrata - zameni sirinu i visinu ukoliko vrata trebaju da se postave vertikalno (horizontalno su po default)
+        private double GetDoorHeight()
+        {
+            if (MapObjectDoorOrientation == MapObjectDoorOrientation.TOP || MapObjectDoorOrientation == MapObjectDoorOrientation.BOTTOM)
+                return DOOR_HEIGHT;
+            else return DOOR_WIDTH;
+        }
+
+        private double CalculateDoorX()
+        {
+            switch (MapObjectDoorOrientation)
+            {
+                case MapObjectDoorOrientation.LEFT:
+                    return MapObjectMetrics.X - GetDoorWidth() + MapObject.RECTANGLE_STROKE_THICKNESS;
+                case MapObjectDoorOrientation.RIGHT:
+                    return MapObjectMetrics.X + MapObjectMetrics.Width - MapObject.RECTANGLE_STROKE_THICKNESS;
+                default:
+                    return CalculateXShifted();
+            }
+        }
+
+        private double CalculateDoorY()
+        {
+            switch (MapObjectDoorOrientation)
+            {
+                case MapObjectDoorOrientation.TOP:
+                    return MapObjectMetrics.Y - GetDoorHeight() + MapObject.RECTANGLE_STROKE_THICKNESS;
+                case MapObjectDoorOrientation.BOTTOM:
+                    return MapObjectMetrics.Y + MapObjectMetrics.Height - MapObject.RECTANGLE_STROKE_THICKNESS;
+                default:
+                    return CalculateYShifted();
+            }
+        }
```

Comment:
### Comment

Default and other cases are not at the same level of abstraction.

### Proposal

```sh
        private double CalculateDoorY()
        {
            switch (MapObjectDoorOrientation)
            {
                case MapObjectDoorOrientation.TOP:
                    return CalculateDoorTopForY();
                case MapObjectDoorOrientation.BOTTOM:
                    return CalculateDoorBottomForY();
                default:
                    return CalculateYShifted();
            }
        }

       private double CalculateDoorTopForY(){
           return MapObjectMetrics.Y - GetDoorHeight() + MapObject.RECTANGLE_STROKE_THICKNESS;
       }

       private double CalculateDoorBottomForY(){
           return MapObjectMetrics.Y + MapObjectMetrics.Height - MapObject.RECTANGLE_STROKE_THICKNESS;
       }

```

I give a proposal for Y but, it is the same proposal for X.

Reply:
I agree, it will be refactored.